### PR TITLE
spline #1034 Consumer API :: Data Source :: multi-value `append` filter

### LINKED ISSUE #1034: Consumer API :: Data Source :: multi-value `append` filter
According to AbsaOSS/spline-ui#179 the filter `append` on a data-source resource should accept any combination of values:
- `true` (append)
- `false` (overwrite)
- `null` (no writes / read-only)

### PR DESCRIPTION
**Pull request by wajda**
Consumer API: Allow for using multi-value filter on the write mode (`append`) property

### CODE REVIEW

File: consumer-rest-core/src/main/scala/za/co/absa/spline/consumer/rest/controller/DataSourcesController.scala [hunk 1]
```diff
@@ -68,8 +68,8 @@ class DataSourcesController @Autowired()(
     @ApiParam(value = "Text to filter the results")
     @RequestParam(value = "searchTerm", required = false) searchTerm: String,
 
-    @ApiParam(value = "Write mode (true - append, false - overwrite")
-    @RequestParam(value = "append", required = false) append: java.lang.Boolean,
+    @ApiParam(value = "Write mode. Comma separated values of: `true` - append, `false` - overwrite, empty string - read only")
+    @RequestParam(value = "append", required = false) append: String,
```
Comment (cerveada) [inline consumer-rest-core/src/main/scala/za/co/absa/spline/consumer/rest/controller/DataSourcesController.scala:72]: It seems weird that `append = ""` means that it's read. Wouldn't it be better to have three booleans for each or something like enum `APPEND`, `OVERWRITE` and `READ`?

Comment (wajda) [reply consumer-rest-core/src/main/scala/za/co/absa/spline/consumer/rest/controller/DataSourcesController.scala:72]: This API is a subject for a redesign, I don't want to spend much time on it.

File: consumer-rest-core/src/main/scala/za/co/absa/spline/consumer/rest/model/WriteModeQuery.scala [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.spline.consumer.rest.model
+
+object WriteModeQuery {
+  def parse(str: String): Array[Option[Boolean]] = {
+    str.split(",").map(s => {
```
Comment (cerveada) [inline consumer-rest-core/src/main/scala/za/co/absa/spline/consumer/rest/model/WriteModeQuery.scala]:
```suggestion
    str.split(",").map(_.trim).map(s => {
```
In rare case, someone sends a space there...

File: consumer-rest-core/src/main/scala/za/co/absa/spline/consumer/rest/controller/DataSourcesController.scala (hunk 1 above)
Comment (cerveada) [reply consumer-rest-core/src/main/scala/za/co/absa/spline/consumer/rest/controller/DataSourcesController.scala:72]: ok

**Review by cerveada — DISMISSED** [review]

File: consumer-rest-core/src/main/scala/za/co/absa/spline/consumer/rest/model/WriteModeQuery.scala (hunk 2 above)
Comment (wajda) [reply consumer-rest-core/src/main/scala/za/co/absa/spline/consumer/rest/model/WriteModeQuery.scala]: That's why one should never save on unit tests :smile: That `trim` used to be there before commit, but somehow vanished at the last moment :smile: :smile: Thank you Adam!

Comment (wajda) [reply consumer-rest-core/src/main/scala/za/co/absa/spline/consumer/rest/model/WriteModeQuery.scala:21]: Ah, no, my bad, it was in another PR

**Pull request merged**